shared/src/commonMain/kotlin/com/vml/tutorial/plantshop/plants/data/DbPlantsDataSource.kt

```diff
@@ -1,34 +1,17 @@
 package com.vml.tutorial.plantshop.plants.data
 
-import app.cash.sqldelight.coroutines.asFlow
-import app.cash.sqldelight.coroutines.mapToList
-import com.vml.tutorial.plantshop.PlantDatabase
-import com.vml.tutorial.plantshop.plants.domain.DbDataSource
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.flow.Flow
+import com.vml.tutorial.plantshop.plants.domain.Plant
+import com.vml.tutorial.plantshop.plants.domain.PlantsDataSource
+import dev.gitlive.firebase.Firebase
+import dev.gitlive.firebase.firestore.firestore
 
-class DbPlantsDataSource(db: PlantDatabase) : DbDataSource {
-    private val queries = db.plantQueries
-    override fun getIds(): Flow<List<Long>> {
-        return queries.getFavoritePlants()
-            .asFlow()
-            .mapToList(Dispatchers.Default)
-    }
-
-    override suspend fun insertToDatabase(id: Int) {
-        queries.insertFavoritePlant(id.toLong())
-    }
-
-    override suspend fun removeFromDatabase(id: Int) {
-        queries.removeFavoritePlant(id.toLong())
-    }
-
-    override suspend fun isIdInDatabase(id: Int): Boolean {
-        try {
-            queries.getFavoritePlantById(id.toLong()).executeAsOne()
-        } catch (err: NullPointerException) {
-            return false
+class DbPlantsDataSource : PlantsDataSource {
```

Comment:
Huhh I think we misunderstood each other successfully :D Sorry If I wasn't clear :) Can we have a LocalDbPlantDataSource as well? it can implement the same interface PlantsDataSource but it will store all the plants locally as we did previously :) I will have another comment in Repository where I will explain when should we use it :)

Reply:
Local plant data source is added